lib/API/v8/Controller/ModuleController.php

```diff
@@ -436,9 +436,18 @@ public function createModuleRecord(Request $req, Response $res, array $args)
             throw $exception;
         }
 
-        if (!empty($beanID = $body['data']['id'])) {
+        if (
+            isset($body['data']['id'])
+            && !empty($body['data']['id'])
+        ) {
+
+            $beanID = $body['data']['id'];
             $bean = \BeanFactory::getBean($moduleName, $beanID);
```

Comment:
If you are in the middle of writing exceptions, maybe you can add some new exceptions here for managing beans.
Like if getBean could be SugarBean|bool, just imagine if returns boolean.....

Reply:
Fair point that there needs to be an exercise to cover more ground for exceptions, but that isn't required here as that would be need to be tackled incrementally. How long is a piece of string would be the phrase I would use for this task.